### PR TITLE
Add quest event group tracking

### PR DESCRIPTION
**Pull request by LukeFZ**

### CODE REVIEW

File: DragaliaAPI.Test/Features/Dungeon/Record/DungeonRecordServiceTest.cs [hunk 1]
```diff
@@ -234,11 +230,10 @@ public async Task GenerateIngameResultData_CallsExpectedMethods()
                 }
             );
 
-        mockQuest.State.Should().Be(3);
```
Comment (SapiensAnatis) [inline DragaliaAPI.Test/Features/Dungeon/Record/DungeonRecordServiceTest.cs:237]: probably better to just remove this assert

File: DragaliaAPI/Features/Quest/QuestCacheService.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        await distributedCache.SetStringAsync(
+            Schema.QuestEventId(playerIdentityService.AccountId, questGroupId),
+            questId.ToString(),
+            QuestEntryCacheOptions
+        );
+    }
+
+    public async Task<int?> GetQuestGroupQuestIdAsync(int questGroupId)
+    {
+        string? questIdString = await distributedCache.GetStringAsync(
```
Comment (SapiensAnatis) [inline DragaliaAPI/Features/Quest/QuestCacheService.cs:25]: worth adding some logging maybe? if we end up having cache issues

File: DragaliaAPI/Services/Game/SavefileService.cs [hunk 3]
```diff
@@ -471,6 +471,18 @@ public async Task Import(LoadIndexData savefile)
...
 
+            this.apiContext.QuestEvents.AddRange(
+                savefile.quest_event_list.MapWithDeviceAccount<DbQuestEvent>(
+                    mapper,
+                    deviceAccountId
+                )
+            );
+
+            this.logger.LogDebug(
+                "Mapping DbSummonTicket step done after {t} ms",
```
[the remark this answers is not in the document]
Comment (LukeFZ) [reply DragaliaAPI/Services/Game/SavefileService.cs]: this is fixed already

File: DragaliaAPI/Features/Quest/QuestService.cs [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            await GenerateBonusDrops(questId, count)
+        ).Select(x => x.ToBuildEventRewardEntityList());
+
+        // Remove at the end so it doesn't get messed up when erroring
+        await questCacheService.RemoveQuestGroupQuestIdAsync(eventGroupId);
+
+        return new AtgenReceiveQuestBonus(questId, count, 1, bonusRewards);
+    }
+
+    private void ResetQuestEventBonus(ref DbQuestEvent questEvent, ref QuestEvent questEventData)
```
[the remark this answers is not in the document]
Comment (LukeFZ) [reply DragaliaAPI/Features/Quest/QuestService.cs]: fixed, the ref for the db entity is also not needed

**Review by SapiensAnatis — DISMISSED** [review]

**Review by SapiensAnatis — APPROVED** [review]

**Pull request merged**